ledger/eval/eval.go

```diff
@@ -1316,6 +1349,10 @@ func (eval *BlockEvaluator) endOfBlock() error {
 			return fmt.Errorf("txn count wrong: %d != %d", eval.block.TxnCounter, expectedTxnCount)
 		}
 
+		if err := eval.validateForPayouts(); err != nil {
```

Comment:
It does "always run" in the sense that it runs during block generation, which runs with `eval.generate && eval.validate`

Reply:
I think that's _true_ but goes against the philosophy that if a block comes in with a cert, you must trust it.  For example, just above the `validateForPayouts()` call, we check `block.TxnCounter`. So, theroetically, a quorum of nodes could change the transaction count, and everybody would just accept it.  Is that a good thing? I don't know, but it is the way things work now. 
I think it's true for signatures too? Do validators check individual signatures if the block has a valid cert? (Maybe they do normally, but not in catch up mode?)

Reply:
Then again, this is outside:
```
	if err := eval.validateAbsentOnlineAccounts(); err != nil {
		return err
	}
```
(and similar)
I think I will move it, and add a comment that tries to convince furthers additions like it to be outside as well.

Reply:
No, there are 82 tests that fail because the proposer is missing.  Presumably it's a lot of code that is testing the ledger but doesn't have a sort "fake agreement" that stuffs in a proposer like I added to DoubleLedger.